apps/mobile/src/components/ContextMenu/ContextMenu.js

```diff
@@ -0,0 +1,217 @@
+        </TouchableOpacity>
+      )}
+      {!widget.view && (
+        <View className='flex-row justify-between items-center content-center'>
+          <Text className='text-sm font-semibold text-foreground'>{title}</Text>
+        </View>
+      )}
+      <View className='flex flex-col justify-center items-center relative'>
+        {/* Need to add the SpecialTopElementRenderer here */}
+      </View>
+      {loading && <Text>{t('Loading...')}</Text>}
+      {listItems.length > 0 && listItems.map(item => <ListItemRenderer key={item.id} item={item} rootPath={rootPath} groupSlug={groupSlug} handleWidgetPress={handleWidgetPress} />)}
+    </View>
+  )
+}
+
+function ListItemRenderer ({ item, rootPath, groupSlug, handleWidgetPress }) {
+  const { t } = useTranslation()
+  const itemTitle = item.title
+  const itemUrl = makeWidgetUrl({ widget: item, rootPath, groupSlug, context: 'group' })
+
+  return (
+    <TouchableOpacity
+      key={item.id + itemTitle}
+      onPress={() => handleWidgetPress(item)}
+      className='flex-row items-center ml-8 h-12 py-2 gap-2 content-center border-b border-foreground/20'
+    >
+      <View className='w-5'><WidgetIconResolver widget={item} className='mr-2' /></View>
+      <Text className='text-sm text-primary-accent'>{itemTitle}</Text>
+    </TouchableOpacity>
+  )
+}
+
+function SpecialTopElementRenderer ({ widget, group }) {
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+  const hasResponsibility = useHasResponsibility({ forCurrentGroup: true, forCurrentUser: true })
+  const canAddMembers = hasResponsibility(RESP_ADD_MEMBERS)
+
+  if (widget.type === 'members' && canAddMembers) {
+    return (
+      <TouchableOpacity
+        onPress={() => navigation.navigate('Group Settings', { screen: 'Invite' })}
+        className='px-4 py-2 bg-primary rounded-md'
+      >
+        <Text className='text-sm font-medium text-white'>{t('Add Members')}</Text>
+      </TouchableOpacity>
+    )
+  }
+
+  if (widget.type === 'about') {
+    return (
+      <View className='mb-4'>
+        <Text className='text-sm text-gray-600 mb-2'>{group.purpose}</Text>
+        <Text className='text-sm text-gray-600'>{group.description}</Text>
+      </View>
+    )
+  }
+}
+
+function ContextWidgetList ({ contextWidgets, groupSlug, rootPath }) {
+  return (
+    <ScrollView className='p-2'>
+      {contextWidgets.map(widget => (
+        <View key={widget.id} className='mb-1'>
+          <ContextMenuItem
+            widget={widget}
+            groupSlug={groupSlug}
+            rootPath={rootPath}
+          />
+        </View>
+      ))}
+    </ScrollView>
+  )
+}
+
+function ContextHeader ({ group }) {
+  const { t } = useTranslation()
+
+  const isPublic = group?.slug === PUBLIC_CONTEXT_SLUG
+  const isMyContext = group?.slug === MY_CONTEXT_SLUG
+  const isAllContext = group?.slug === ALL_GROUPS_CONTEXT_SLUG
+  const isGroupContext = !isAllContext && !isMyContext && !isPublic
+  return null
```

Comment:
You see what I mean about that isPublic, isMyContext, etc thing repeating everywhere. This is why we want this stuff on the presenter and to think it through at that level.